templates/sql-ec2-tstore/hazelcast.xml

```diff
@@ -0,0 +1,122 @@
+            </memory-tier>
+            <disk-tier enabled="true" device-name="default-tiered-store-device"/>
+        </tiered-store>
+    </map>
+    <map name="JoinBenchmark">
+        <in-memory-format>NATIVE</in-memory-format>
+    </map>
+
+    <local-device name="default-tiered-store-device">
+        <base-dir>/home/ubuntu/tiered-store</base-dir>
+    </local-device>
+
+    <native-memory enabled="true" allocator-type="POOLED">
+        <capacity unit="GIGABYTES" value="16"/>
+    </native-memory>
+
+    <jet enabled="true" resource-upload-enabled="false">
+		    <!--    <cooperative-thread-count>8</cooperative-thread-count> -->
+        <!-- time spacing of flow-control (ack) packets -->
+        <flow-control-period>100</flow-control-period>
+        <!-- number of backup copies to configure for Hazelcast IMaps used internally in a Jet job -->
+        <backup-count>1</backup-count>
+        <!-- the delay after which auto-scaled jobs will restart if a new member is added to the
+             cluster. The default is 10 seconds. Has no effect on jobs with auto scaling disabled -->
+        <scale-up-delay-millis>10000</scale-up-delay-millis>
+        <!-- Sets whether Lossless Job Restart is enabled for the node. With
+             Lossless Restart, Jet persists job snapshots to disk automatically
+             and you can restart the whole cluster without losing the jobs and
+             their state.
+
+             This feature requires Hazelcast Enterprise and is implemented on top of the
+             Persistence feature. Therefore you should enable and configure Persistence,
+             especially the base directory where to store the recovery files.
+         -->
+        <lossless-restart-enabled>false</lossless-restart-enabled>
+        <!-- Sets the maximum number of records that can be accumulated by any single
+             Processor instance.
+             Operations like grouping, sorting or joining require certain amount of
+             records to be accumulated before they can proceed. You can set this option
+             to reduce the probability of OutOfMemoryError.
+
+             This option applies to each Processor instance separately, hence the
+             effective limit of records accumulated by each cluster member is influenced
+             by the vertex's localParallelism and the number of jobs in the cluster.
+
+             Currently, maxProcessorAccumulatedRecords limits:
+                - number of items sorted by the sort operation
+                - number of distinct keys accumulated by aggregation operations
+                - number of entries in the hash-join lookup tables
+                - number of entries in stateful transforms
+                - number of distinct items in distinct operation
+
+             Note: the limit does not apply to streaming aggregations.
+         -->
+        <max-processor-accumulated-records>9223372036854775807</max-processor-accumulated-records>
```

Comment:
nit: you risk OOM with such setting in case of bad query, but that may not be a big concern

Reply:
Another setting based on precedent (rather than thoughtfully chosen): https://github.com/hazelcast/hazelcast-simulator/blob/5d786228966d76863fe8accbdf222f7650364339/templates/sql-ec2/hazelcast.xml#L91

My template is basically a modified version of `sql-ec2` that uses tiered-storage OOTB.